### PR TITLE
feat: Adding web-threejs library to sdk

### PR DESCRIPTION
**Pull request by Disfractal**
## Description

feat: Added the new Three.js module to the web sdk
fix: Updated root .gitignore for new files

## Related Ticket (for Inworld.ai developers)

https://inworldai.atlassian.net/jira/software/c/projects/CORE/boards/8?quickFilter=26&selectedIssue=CORE-4705

### CODE REVIEW

File: packages/web-threejs/src/innequin/Innequin.ts [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+        this.animationGestures.push(animationGesture);
+      }
+    }
+    // console.log('Animations Loaded');
+    this.loadVismeMaterials();
+  }
+
+  onLoadComplete() {
+    this.onLoad(this.config);
+    console.log('Innequin - Character Loaded.');
```
Comment (tshashkova) [inline packages/web-threejs/src/innequin/Innequin.ts:220]: Maybe you add debug option for all console.log which are present in package
What do you think?
It's nice to understand the loading progress, but it should be up to developer needs

File: packages/web-threejs/src/innequin/animator/utils/InnequinPhonemesToViseme.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { AdditionalPhonemeInfo } from '@inworld/web-core';
+
+import { InnequinPhonemesToVisemesMap } from './InnequinPhonemesToVisemesMap';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (tshashkova) [inline packages/web-threejs/src/innequin/animator/utils/InnequinPhonemesToViseme.ts:5]: Let's check that eslint-disable is used only it's impossible to compile application without these comments
For example, seems here  you just don't use EXPRESSION_FACTOR and EXPRESSION_FACTOR. And this variables can be removed at all

File: packages/web-threejs/src/rpm/RPM.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  }
+
+  setPhonemes(phonemes: AdditionalPhonemeInfo[]) {
+    if (this.animator) {
+      this.animator.setPhonemes(phonemes);
+    }
+  }
+
+  updateFrame(delta: number) {
+    if (this.animator && this.animator.animatorReady) {
```
Comment (tshashkova) [inline packages/web-threejs/src/rpm/RPM.ts:157]: Just simplify
`if (this.animator?.animatorReady) {}`

**Review by tshashkova — CHANGES_REQUESTED** [review]
Add new package to ISSUE_TEMPLATE files too

File: packages/web-threejs/package.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    "docs:generate": "npx typedoc"
+  },
+  "devDependencies": {
+    "@types/react": "^18.0.28",
+    "@types/react-dom": "^18.0.11",
+    "@types/three": "^0.157.0",
+    "@types/uuid": "^9.0.0",
+    "eslint-config-prettier": "^8.8.0",
+    "eslint-plugin-markdown": "^3.0.0",
+    "eslint-plugin-prettier": "^4.2.1",
```
[the remark this answers is not in the document]
Comment (Disfractal) [reply packages/web-threejs/package.json:50]: Removed all dependencies common

File: packages/web-threejs/.npmignore [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+/src
+/proto
+*/**/node_modules/*
+jest.config.json
+/tsconfig.json
+/.release-it.json
+*/**/yarn.lock
+*/**/package-lock.json
+*.log
+EXTENSIONS.md
```
[the remark this answers is not in the document]
Comment (Disfractal) [reply packages/web-threejs/.npmignore:12]: Is it the imgs folder and readme? I'm not sure what a pack for NPM would consist of.

File: packages/web-threejs/src/innequin/Innequin.ts (hunk 1 above)
Comment (Disfractal) [reply packages/web-threejs/src/innequin/Innequin.ts:220]: Ya I thought about this as well. Maybe an enviroment var which is false by default.

File: packages/web-threejs/.npmignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (tshashkova) [reply packages/web-threejs/.npmignore:12]: README, yes
And images if it's required part of package
So, I believe everything is correct from your side, JIC

**Comment by tshashkova** [comment]
@Disfractal If you're not ready to add tests right now, it's better to exclude them at all for your package
I mean not pass option ( --passWithNoTests), just remove **yarn test** command call from your package (release-it)

**Comment by Disfractal** [comment]
> @Disfractal If you're not ready to add tests right now, it's better to exclude them at all for your package I mean not pass option ( --passWithNoTests), just remove **yarn test** command call from your package (release-it)

Originally yarn test wasn't included in the package.json but when I added it with --passWithNoTests I got this error:
> @inworld/web-threejs
error Command "test" not found. Did you mean "jest"?

**Review by tshashkova — APPROVED** [review]

**Pull request merged**